Zend/zend_max_execution_timer.c

```diff
@@ -33,6 +33,8 @@
 # define sigev_notify_thread_id _sigev_un._tid
 # endif
 
+_Thread_local int is_handler_set = 0; // whether we have setup a timer in this thread
```

Comment:
While at it, please make this a bool instead of an int, as this actually acts as a boolean flag.
Also, I think that _Thread_local is C11 while PHP uses C99 atm?

Reply:
Yes, we officially only use C99. C11 feature needs to be added to the portability header, similarly to the ``zend_atomic`` stuff we do.

Reply:
Oh, good catch! I'll take a look at `zend_atomic` and see how to do this properly.

Reply:
I've added `ZEND_THREAD_LOCAL` to `zend_portability.h`, though I'm not 100% sure I got it right.

Reply:
Maybe it's me misunderstanding something, but I think you should be using `ZEND_TLS`? That seems to do the right thing. Then you can remove your own definition.

Reply:
Hmm, interesting. `ZEND_TLS` comes from `TSRM_TLS` which is defined as `__thread`, which is specific to GNU C, while `_Thread_local` is C11.

Nice find! It still does what I need, so I'll use it!